firmware/applications/SignBuddy/imu.c

```diff
@@ -0,0 +1,119 @@
+#include "imu.h"
+
+#include "board.h"
+#include "common.h"
+#include "i2c.h"
+#include "logger.h"
+#include "bno055.h"
+
+#define PROCESS_PERIOD_MS    1000
+
+typedef struct {
+  uint32_t                last_ticks;
+  i2c_t                   i2c_instance;
+  struct   bno055_t       bno055;
+  struct   bno055_accel_t bno055_accel_xyz;
+  struct   bno055_mag_t   bno055_mag_xyz;
+  struct   bno055_gyro_t  bno055_gyro_xyz;
+} state_t;
+
+static state_t s;
+
+static void hw_init(void)
+{
+  IMU_I2C_CLK_EN();
+  IMU_I2C_GPIO_CLK_EN();
+
+  LL_GPIO_InitTypeDef gpio_config = { 0 };
+  gpio_config.Pin = LL_GPIO_PIN_6 | LL_GPIO_PIN_7;
+  gpio_config.Mode = LL_GPIO_MODE_ALTERNATE;
+  gpio_config.Speed = LL_GPIO_SPEED_FREQ_VERY_HIGH;
+  gpio_config.OutputType = LL_GPIO_OUTPUT_OPENDRAIN;
+  gpio_config.Pull = LL_GPIO_PULL_NO;
+  gpio_config.Alternate = LL_GPIO_AF_1;
+  LL_GPIO_Init(GPIOB, &gpio_config);
+
+  LL_I2C_InitTypeDef i2c_config = { 0 };
+  LL_I2C_EnableAutoEndMode(IMU_I2C);
+  LL_I2C_DisableOwnAddress2(IMU_I2C);
+  LL_I2C_DisableGeneralCall(IMU_I2C);
+  LL_I2C_EnableClockStretching(IMU_I2C);
```

Comment:
Do we need to call these functions here and in i2c_init?

Reply:
no we don't. good catch. I think we can put that into the i2c init